File: SQLite数据库操作-0C/SQLite数据库操作-0C/ViewController.m
//
//  ViewController.m
//  SQLite数据库操作-0C
//
//  Created by 邱学伟 on 16/10/24.
//  Copyright © 2016年 邱学伟. All rights reserved.
//

#import "ViewController.h"
#import "SQLiteManager.h"
#import "User.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    /**
    for (NSInteger i = 1; i < 8; i++) {
        NSString *name = [NSString stringWithFormat:@"name_%ld",i];
        NSInteger age = arc4random_uniform(18) + i + 5;
        NSString *icon = [NSString stringWithFormat:@"http://qiuxuewei.com/icon_%ld.png",i];
        NSLog(@"arc4random_uniform(2) : %u",arc4random_uniform(2));
//        [[SQLiteManager shareInstance] execSQL:@"DELETE FROM 't_User'"];
        if (arc4random_uniform(2) == 1) {
            User *USER = [[User alloc] initWithName:name andAge:age andIcon:icon];
            if ([USER insertSelfToDB]) {
                NSLog(@"对象 %ld 插入成功!",i);
            }else{
                NSLog(@"对象 %ld 插入失败!",i);
            }
        }else{
            NSDictionary *dict = @{
                                   @"name":name,
                                   @"age":[NSNumber numberWithInteger:age],
                                   @"icon":icon
                                   };
            User *USER = [[User alloc] initWithDict:dict];
            if ([USER insertSelfToDB]) {
                NSLog(@"对象 %ld 插入成功!",i);
            }else{
                NSLog(@"对象 %ld 插入失败!",i);
            }
        }
    }
    */
    
//    [User allUserFromDB];
    
    [self updateIcon];
    
    
}
-(void)updateIcon{
    //更新对应的SQL语句
    NSString *SQL = [NSString stringWithFormat:@"UPDATE 't_User' SET icon='%@' WHERE name = '%@'",@"http://qiuxuewei.com/newIcon.png",@"name_6"];
    if ([[SQLiteManager shareInstance] execSQL:SQL]) {
        NSLog(@"对应数据修改成功");
    }
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SQLite数据库操作-Swift/SQLite数据库操作-Swift/ViewController.swift
//
//  ViewController.swift
//  SQLite数据库操作-Swift
//
//  Created by 邱学伟 on 16/10/24.
//  Copyright © 2016年 邱学伟. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
       
    }
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
//        for i in 1...7 {
//            let name = "name_\(i)"
//            let age = arc4random_uniform(18).hashValue + i
//            let icon = "http://qiuxuewei.com/icon\(i).png"
//            let user : User = User(name: name, age: age, icon: icon)
//            if user.insertSelfToDB() {
//                print("第 \(i) 个用户插入成功!")
//            }
//        }
        
        //修改name_6头像为newIocn
//        changeIcon(newIcon: "http://newIcon.png")
        
        //取出用户表所有数据
        let userArr = User.allUserFromDB()
        print(userArr)
    }
    
    func changeIcon(newIcon : String) {
        //包装修改头像的SQL语句
        let changeIconSQL = "UPDATE 't_User' SET icon='\(newIcon)' WHERE name='name_6'"
        if SQLiteManager.shareInstance().execSQL(SQL: changeIconSQL) {
            print("name_6 头像修改成功!")
        }
    }


}

